### ISSUE TITLE
Mini LLM training , time 72.2 sec , val_loss = 1.901

### ISSUE THREAD

**Issue by vegansquirrel** [vukrosic/ultimate-ai-research-and-engineering-course#3]
<img width="564" height="157" alt="Image" src="https://github.com/user-attachments/assets/4a1ac2f3-9cef-4bbb-b4c8-d677d1c106e7" />

Mini LLM validation Loss (<=2) in record time .

weight Decay = 0.001
learning rate to 0.005
T_max set to steps_per_epochs * epochs .
criterion =  nn.CrossEntropyLoss
